src/test/HostActivationTests/NativeHosting/HostContext.cs

```diff
@@ -224,166 +160,368 @@ public void GetDelegate_Multiple(string checkProperties)
                     result.Should()
-                        .GetRuntimePropertiesIncludes(LogPrefix.App, sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.GetRuntimePropertiesIncludes(LogPrefix.Secondary, sharedState.ConfigPropertyName, sharedState.ConfigPropertyValue)
-                        .And.GetRuntimePropertiesExcludes(LogPrefix.Secondary, sharedState.AppPropertyName)
-                        .And.HavePropertyMock(sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.NotHavePropertyMock(sharedState.ConfigPropertyName);
+                        .InitializeContextForConfig(appOrConfigPath)
+                        .And.CreateDelegateMock();
                     break;
-                case CheckProperties.GetActive:
+                case Scenario.Mixed:
                     result.Should()
-                        .FailToGetRuntimePropertyValue(LogPrefix.App, sharedState.AppPropertyName, HostInvalidState)
-                        .And.FailToGetRuntimePropertyValue(LogPrefix.App, sharedState.ConfigPropertyName, HostInvalidState)
-                        .And.GetRuntimePropertyValue(LogPrefix.Secondary, sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.FailToGetRuntimePropertyValue(LogPrefix.Secondary, sharedState.ConfigPropertyName, HostPropertyNotFound)
-                        .And.HavePropertyMock(sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.NotHavePropertyMock(sharedState.ConfigPropertyName);
+                        .InitializeContextForApp(appOrConfigPath)
+                        .And.ExecuteAssemblyMock(appOrConfigPath, new string[0]);
                     break;
-                case CheckProperties.GetAllActive:
+                case Scenario.NonContextMixed:
                     result.Should()
-                        .FailToGetRuntimeProperties(LogPrefix.App, HostInvalidState)
-                        .And.GetRuntimePropertiesIncludes(LogPrefix.Secondary, sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.GetRuntimePropertiesExcludes(LogPrefix.Secondary, sharedState.ConfigPropertyName)
-                        .And.HavePropertyMock(sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.NotHavePropertyMock(sharedState.ConfigPropertyName);
+                        .ExecuteAssemblyMock(appOrConfigPath, new string[0]);
                     break;
-                default:
-                    throw new Exception($"Unknown option: {checkProperties}");
+            }
+
+            if (existingValue == null)
+            {
+                result.Should()
+                    .InitializeSecondaryContext(propertyCompatConfig, CoreHostDifferentProperties)
+                    .And.HaveStdErrContaining($"The property [{propertyName}] is not present in the previously loaded runtime");
```

Comment:
Can we also validate the return code from the call? This should return the special code...

Reply:
`InitializeSecondaryContext` checks the return code (that the nativehost test exe prints)?